File: 58-length-of-last-word/length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {

    // --simple 3 line 

    // s=s.trim().split(' ');
    // let x = s[s.length-1]
    // return x.length;
    s=s.toLowerCase();
    let n = s.length-1;
    let c = 0;

    while(n>=0){

        if(s[n]!=' ') c++; 
        // if(s[n]>='a' && s[n]<='z') c++;   --can also use this 
        else if(c>0) break;

        n--;
    }
    return c;


};

File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {

        int n = s.length()-1;
        int c = 0;

        while(n>=0) {

            if(s.charAt(n) != ' ')  c++;

            else if(c>0)  break;


            n--;
    
        }
        return c;
        
    }
}